airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/jinja.py

```diff
@@ -74,20 +82,29 @@ def eval(self, input_str: str, config: Config, default: Optional[str] = None, **
             if isinstance(input_str, str):
                 result = self._eval(input_str, context)
                 if result:
-                    return self._literal_eval(result)
+                    return self._literal_eval(result, valid_types)
             else:
                 # If input is not a string, return it as is
                 raise Exception(f"Expected a string. got {input_str}")
         except UndefinedError:
             pass
         # If result is empty or resulted in an undefined error, evaluate and return the default string
-        return self._literal_eval(self._eval(default, context))
+        return self._literal_eval(self._eval(default, context), valid_types)
 
-    def _literal_eval(self, result):
+    def _literal_eval(self, result, valid_types: Optional[Tuple[Type]]):
         try:
-            return ast.literal_eval(result)
+            evaluated = ast.literal_eval(result)
         except (ValueError, SyntaxError):
             return result
+        if not valid_types or (valid_types and isinstance(evaluated, tuple(valid_types))):
```

Comment:
Probably nit: Why cast `valid_types` as a tuple? Isn't it already a tuple?

Reply:
It was a List - changed it to a Tuple and removed the cast.